src/components/editor/document-bar/permissions/permission-modal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react'
+import { Alert, Modal } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { getUserById } from '../../../../api/users'
+import { CommonModal } from '../../../common/modals/common-modal'
+import { ShowIf } from '../../../common/show-if/show-if'
+import { UserAvatar, UserAvatarProps } from '../../../common/user-avatar/user-avatar'
+import { GroupMode, PermissionGroupEntry } from './permission-group-entry'
+import { PermissionList } from './permission-list'
+
+export interface PermissionsModalProps {
+  show: boolean,
+  onChangeShow: (newShow: boolean) => void
+}
+
+export interface Principal {
+  id: string
+  name: string
+  photo: string
```

Comment:
You could strip down the attributes by extending `UserResponse`.

```suggestion
export interface Principal extends UserResponse {
```

If you apply this change, you need to add the `provider` attribute in the `addUser` method as it is defined in the response as well.

Reply:
Objection!
A principal ist not an api response. Please don't extend stuff just because they have similar fields. Try to maintain a logic structure.

Reply:
Well, I see your point.

I had the idea of extending UserResponse, because the other attributes except `canEdit` are in fact fetched from the API which returns the response object.

Maybe we should define some common User-attributes which are used by the API response, by the Principal objects, by the Avatar-objects? I don't think it makes much sense to copy the exact same attributes into each new interface that has something to do with users.

Reply:
That would be a better solution

Reply:
I'd suggest working that out in separate PR